### DISCUSSION TITLE
Numbering and font on linux

### DISCUSSION THREAD

**Discussion by dogarrowtype** [hlky/stable-diffusion-webui#220] [Feature suggestions]
Hi, awesome fork.
On linux at least, when saving individual images, webui.py starts at a negative number. (-0001, 00000, 00001, 00002, etc). Saving a grid is fine.
And, when the font "arial.ttf" is not present on the system (most linux distros don't package arial), the script throws an error (doesn't crash though) and refuses to show any images on the gradio ui. It's throwing an error in image_grid() that is easily fixed by replacing with "data/DejaVuSans.ttf".

I'm happy to provide more debugging info if you need it, or test on linux.

**Comment by hlky** [comment]
I think this might have been fixed, there is `get_font(fontsize):` now which tries to load `fonts = ["arial.ttf", "DejaVuSans.ttf"]`

The number should have been fixed too

leaving open for now and moving to dev repo

**Reply by dogarrowtype** [reply]
Both of these issues have been fixed, thanks!

**Comment by JoshuaKimsey** [comment]
Out of curiosity, instead of calling a specific font, is it possible to just call a general font-family, like sans-serif, instead? That way there should be no issues with whatever fonts users have installed?

**Reply by hlky** [reply]
Unfortunately I don't think it works like that, the font is used to put the text on the prompt matrix images so needs the font file.

**Comment by oc013** [comment]
Please see here where I implemented a fix in a docker image https://github.com/hlky/stable-diffusion/commit/9bebf683f0be3f83b6b35ae818943f31b33c9c18  You should be able to do the same thing on your actual system.

This basically "installs" the font and I believe should be where PIL will search for the ttf file on a linux system.

 The alternative would be to extend the list to include the path `fonts = ["data/DejaVuSans.ttf", "arial.ttf", "DejaVuSans.ttf"]`
